### PR TITLE
use a function in python

### PR DESCRIPTION
**Pull request by bepvte**
global variables in python are far slower then locals
my testing shows this speeds up `loops` by 2x

### CODE REVIEW

**Comment by 0xSaltyHash** [comment]
yeah testing this on my wsl2 with python3.12 gave me 45.370 S
image

**Comment by james598keen** [comment]
Hello,
You're right! Local variables in Python are indeed faster than global variables because local variable access involves fewer lookup steps. This optimization can significantly speed up loops, potentially doubling the performance as you've noticed.

Best Regards,
James Keen

File: fibonacci/py/code.py [hunk 1]
```diff
@@ -7,8 +7,11 @@ def fibonacci(n):
...
   return fibonacci(n-1) + fibonacci(n-2)
 
-u = int(sys.argv[1])
-r = 0
-for i in range(u):
-  r += fibonacci(i)
-print(r)
+def main():
+    u = int(sys.argv[1])
+    r = 0
```
Comment (oriontvv) [inline fibonacci/py/code.py:12]: Pythonic way here is to use builtin `sum` function. And type hints would be nice to have.

Comment (bddicken) [reply fibonacci/py/code.py:12]: The idea is to implement the "same" way across all implementations, so will not be adding sum at this time.

**Comment by bddicken** [comment]
Wow cool, thanks for the contribution.

**Pull request merged**